### ISSUE TITLE
implement a pull request landing bot

### ISSUE THREAD

**Issue by mikesherov** [jquery/esprima#1018]
Considering our complex commit message guide, having a bot actually land patches will reduce errors in commit messages and take the burden out of landing the patch. We should investigate if existing solutions are already out there that meet our requirements as a prereq.

**Comment by ariya** [comment]
As a side note, it is likely a good for the landing bot to also run a more extensive test suite (e.g. #1023). However, it's perfectly fine if the first stage is only to land the PR as is.

**Comment by ariya** [comment]
@mikesherov @scottgonzalez Is there anything I can learn from the CLA bot to implement this one?

**Comment by scottgonzalez** [comment]
Perhaps. I'd be happy to go over how the CLA bot works and discuss possible implementations for this new bot.

**Comment by ariya** [comment]
@scottgonzalez Is the current CLA bot an instance of https://github.com/jquery/jquery-license?

**Comment by scottgonzalez** [comment]
yes

**Comment by ariya** [comment]
Unfortunately I don't think I am able to implement this in the near future. Unless someone volunteers, I think I will close this as WontFix. We can reopen it again when the opportunity presents itself.

**Comment by scottgonzalez** [comment]
I'm fairly busy right now, but building some generic tooling around PRs is something I've been considering for a while. If you want to lay out how you envision this working, I might work on this. But don't wait on me, since it might be a while.

**Comment by ariya** [comment]
The only thing I had in mind is as follows. When a collaborator poke the bot, e.g. "Hey landingbot, this is good to merge", then the bot will react: rebase the pull request branch, modify the commit log to mention the pull request, merge it to the intended target branch, push the branch, and close the pull request.
If any of the steps fail, e.g. conflict during rebase, then it should post the error message.

I believe this will be useful for many other jQuery Foundation projects anyway, so perhaps we should move the brainstorming to a more proper place!

**Comment by ariya** [comment]
Moving the discussion to https://github.com/jquery/jquery-license/issues/59.
